File: Module 19-Sorting&algo/03_Pushzerotoend.cpp
#include<iostream>
#include<vector>
using namespace std;
int main(){
    vector<int> arr = {4,0,1,2,0,2,0,2,0,3,0,5}; //we want to maintain the order of orginal number 
    int n = arr.size();
    for (int i = 0; i < n - 1; i++) {
        bool flag = true;
        for (int x = 0; x < n - 1 - i; x++) {
            if (arr[x] == 0 and arr[x + 1]!=0) {
                swap(arr[x], arr[x + 1]); // one-line swap
                flag = false;
            }
        }
        if (flag) break;
    }

    // Print sorted array
    for (int num : arr) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}

File: Module 19-Sorting&algo/03_Pushzerotoend.py
arr = [4, 0, 1, 2, 0, 2, 0, 2, 0, 3, 0, 5]  # we want to maintain the order of original numbers
n = len(arr)

for i in range(n - 1):
    flag = True
    for x in range(n - 1 - i):
        if arr[x] == 0 and arr[x + 1] != 0:
            arr[x], arr[x + 1] = arr[x + 1], arr[x]  # one-line swap
            flag = False
    if flag:
        break

# Print sorted array
for num in arr:
    print(num, end=" ")
print()
